pkg/provision/engine.go

```diff
@@ -577,6 +593,13 @@ func (e *NativeEngine) contract(ctx context.Context, dl *gridtypes.Deployment, n
 	}
 	ctx = withContract(ctx, contract.ContractType.NodeContract)
 
+	rent, err := sub.GetNodeRentContract(e.nodeID)
+	if err != nil && !errors.Is(err, substrate.ErrNotFound) {
+		return nil, fmt.Errorf("failed to check node rent state")
+	}
+
```

Comment:
We can reach this line in case of error is ErrNotFound,  also I don't understand the logic in error == nil && rent != 0 and how it can be affected with ErrNotFound :(

Reply:
as you said, this only handles any error during the "contract get" call above that is ErrNotFound. so basically return if we have an error that is not ErrNotFound. 

The thing is, if there are NO rent contract for the node the call normally returns `0` but imho this is not correct it instead should return NotFound error. So i am trying to be a little bit more defensive in the code by handling the error correctly.

But anyway, if no error or ErrNotFound error was returned then (in both cases) 0 means No rent, and any other value is rented, hence later i make sure that the node is only marked as rented if and only if there is no error and the rent contract has non zero value.